pkg/cmd/task/task_log.go

```diff
@@ -49,21 +49,34 @@ func newCmdTaskLog(ctx api.Context) *cobra.Command {
 
 			if len(tasks) == 0 {
 				return fmt.Errorf("no task ID found containing '%s'", args[0])
-			} else if len(tasks) > 1 {
-				return fmt.Errorf("found more than one task with the same name: %v", tasks)
 			}
 
-			logClient := logs.NewClient(pluginutil.HTTPClient(""), ctx.Out())
-			if output != "short" {
-				output = "short"
-				ctx.Logger().Info(`Task logs don't support output options. Defaulting to "short"...`)
-			}
-			opts := logs.Options{
-				Follow: follow,
-				Format: output,
-				Skip:   -1 * lines,
+			// Only one task matching or multiple tasks but no follow.
+			if len(tasks) == 1 || follow == false {
+				for _, task := range tasks {
+					if len(tasks) > 1 {
+						fmt.Fprintln(ctx.Out(), fmt.Sprintf("===> %s <===", task))
+					}
+					logClient := logs.NewClient(pluginutil.HTTPClient(""), ctx.Out())
+					if output != "short" {
+						output = "short"
+						ctx.Logger().Info(`Task logs don't support output options. Defaulting to "short"...`)
+					}
+					opts := logs.Options{
+						Follow: follow,
+						Format: output,
+						Skip:   -1 * lines,
+					}
+					err = logClient.PrintTask(task, file, opts)
+					if err != nil {
+						return err
```

Comment:
Instead of immediately returning we could only log the error here so that we potentially still get logs from other tasks. At least we decided to do that in other places of the CLI.

Reply:
OK, will change that when working on following multiple tasks.